src/electron/shell/electron-toolbar.service.ts

```diff
@@ -0,0 +1,67 @@
+import {Injectable} from "@angular/core";
+import {remote, Menu, MenuItem} from "electron";
+import * as _ from "lodash";
+
+import {FileToolbarAction, FileToolbarService} from "../../duckling/shell/file-toolbar.service";
+
+/**
+ * Electron implementation of the FileToolbarService used to handle the menu at the top of the
+ * window.
+ */
+@Injectable()
+export class ElectronToolbarService extends FileToolbarService {
+    constructor() {
+        super();
+
+        this.addAction({
+            menuPath : ["File"],
+            label : "Close Project",
+            shortcut: "CmdOrCtrl+R",
+            callback : () => remote.getCurrentWindow().reload()
+        });
+    }
+
+    bootstrapMenu() {
+        remote.Menu.setApplicationMenu(this.toMenu(this.actions));
+    }
+
+    toMenu(actions : FileToolbarAction []) {
+        let menu = new remote.Menu();
+
+        let [subMenuActions, items] = _.partition(actions, action => action.menuPath.length);
+        let subMenus = _.groupBy(subMenuActions, action => action.menuPath[0]);
```

Comment:
This isn't really obvious what's going on, can we refactor it out into it's own function that describes it?